Tasks/NuGetInstaller/task.json

```diff
@@ -29,6 +30,14 @@
             "helpMarkDown": "The path to the Visual Studio solution file or NuGet packages.config",
             "required": "true"
         },
+		{
+            "name": "connectedServiceName",
+            "type": "connectedService:Generic",
```

Comment:
You're going to introduce a lot of compat issues and upgrade code that you will need to write.  Upgrade code that's also very expensive.

The types have extensibility and are trivial to write.

I would highly recommend holding this till that's in.  Should be quick to add.